### ISSUE TITLE
Document cluster restart procedure

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#1398]
Reported by @robert-s-lee:

https://www.cockroachlabs.com/docs/stop-a-node.html has a documentation on stopping a node at a time. One can infer stopping each of the nodes individually will stop the cluster; however, there should be a documented way to stop and restart the whole cluster or a portion of the cluster.

**Comment by sploiselle** [comment]
@robert-s-lee Just curious, is there a specific use case you see this applying to?

@bdarnell Would this kind of command normally be executed through a process manager? Are things like our zone-config attributes visible to process managers as a way to filter which nodes you want to bring down?

**Comment by jseldess** [comment]
Given that this shouldn't be necessary technically (auto-repair, rolling upgrades, etc.), I asked the same thing in person, and Robert mentioned that it's common practice at some financial institutions to shut down a cluster periodically, though I didn't understand exactly why. I think he mentioned for internal policy reasons or to verify that it can be brought back up successfully if the cluster fails unexpectedly. Interested in getting more clarify from @robert-s-lee, too.

**Comment by bdarnell** [comment]
A process manager would typically use commands like `service cockroachdb stop` or `supervisorctl stop cockroach`, but it's all going to be very deployment-specific. Until we're recommending and documenting a more specific stack, it's not worth mentioning some of the commands that might be used.

Stopping all the nodes is actually a little tricky, because once you've taken down at least two nodes the other nodes may not be able to shut down cleanly (they'll eventually time out and exit, but only after waiting a while). If you're shutting down the whole cluster, you don't want to do anything cockroach-specific. Just `kill -9` everything.

**Comment by robert-s-lee** [comment]
Documented stop and start procedure is normally Incorporated in a run book -- operators use the run book to keep services running 24 x 7.  Enterprises practice disaster recovery procedures per regulatory requirements at fixed intervals per year where apps / databases are stopped and started.  Some companies have long standing practice to restart all services over the weekend.  This basically forces test of automation scripts in  production on weekly basis so that any intentional and unintentional restart during the week known to work.  

If kill -9 is the recommended way ( this is a dapature from other databases ) we should test and document.  Other databases recommend a control shutdown procedure to control recovery time on the restart.

**Comment by sploiselle** [comment]
@robert-s-lee From your experience, would a rolling restart of all of the services meet the need? i.e. all of the individual nodes would be restarted, but it wouldn't require taking the entire cluster down at once

**Comment by bdarnell** [comment]
> Other databases recommend a control shutdown procedure to control recovery time on the restart.

If you only test your restart scripts after a controlled shutdown, you haven't really tested your restart scripts :)

**Comment by jseldess** [comment]
Closing in favor of https://github.com/cockroachdb/docs/issues/1772.
